lib/matplotlib/tests/test_datetime.py

```diff
@@ -399,11 +399,20 @@ def test_matshow(self):
         fig, ax = plt.subplots()
         ax.matshow(...)
 
-    @pytest.mark.xfail(reason="Test for pcolor not written yet")
     @mpl.style.context("default")
     def test_pcolor(self):
-        fig, ax = plt.subplots()
-        ax.pcolor(...)
+        base_date = datetime.date(2020, 1, 1)
+        dates = [base_date + datetime.timedelta(days=i) for i in range(10)]
+        data = np.random.rand(10, 10)
+        data = np.sort(data, axis=0)
+        fig, ax = plt.subplots()
+        c = ax.pcolor(data, cmap='viridis')
```

Comment:
There is no use of dates in this call.

Reply:
Both axes of pcolor require that I pass numerical data, how do you suggest I use dates? I can use datetime in this call, but then I would have to use date2num, which defies the purpose. Not sure how else to implement. This is what I found online:

'In my understanding, you would like to have date formatted tick labels for axes when using 'pcolor'. You cannot directly provide 'pcolor' with datetime vectors as of now.
However, there is a way to use the ' [datenum](https://www.mathworks.com/help/matlab/ref/datenum.html) ' and ' [datetick](https://www.mathworks.com/help/matlab/ref/datetick.html) ' functions to get datetime axes tick labels as illustrated in the example below:
%Creating two example vectors of datetimes, tx and ty
t1 = datetime(2017,9,2,0,0,0);
t2 = datetime(2017,9,30,0,0,0);
tx = t1:5:t2;
ty = tx + day(15);
%Creating an example C matrix for pcolor plot
C = magic(length(tx));
% Using datenum to convert to real numbers for plotting and dateticks for
% date labelling. You can play around with datetick options to suit your needs 
pcolor(datenum(tx), datenum(ty), C);
datetick('y', 'dd-mm-yy',  'keepticks');
datetick('x', 'dd-mm-yy',  'keepticks');'

Reply:
X and Y in fact, _CAN_ take date info directly, no need to call `date2num`...

```python
>>> tx = ty = np.arange('2005-02', '2005-03', dtype='datetime64[D]')
>>> tx, ty = np.meshgrid(tx, ty)
>>> c = np.arange(tx.size).reshape(tx.shape)
>>> plt.pcolor(tx, ty, c)
```

yields
![test](https://github.com/matplotlib/matplotlib/assets/2501846/4ea65355-a861-4f56-83dd-ee3d81c7d752)


While I made no effort to clean up the labels so they were visible/not overlapping, the point stands that they are properly converted as desired/as is the point of these tests.

Reply:
The answer you found is Matlab code about Matlab's plotting, which matplotlib was originally inspired by, but has diverged from significantly since then.